internal/pkg/service/common/etcdclient/etcdclient.go

```diff
@@ -48,22 +48,16 @@ func New(ctx context.Context, proc *servicectx.Process, tel telemetry.Telemetry,
 	logger = logger.WithComponent("etcd.client")
 
 	// Create a zap logger for etcd client
-	encoder := zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig())
 	etcdLogger := zap.New(log.NewCallbackCore(func(entry zapcore.Entry, fields []zapcore.Field) {
 		// Skip debug messages
 		if entry.Level == log.DebugLevel {
 			return
 		}
 
-		// Add component=etcd.client field
-		fields = append(fields, zapcore.Field{Key: "component", String: "etcd.client", Type: zapcore.StringType})
+		var attributes []attribute.KeyValue
+		ctxattr.ZapFieldsToAttrs(fields, &attributes)
```

Comment:
Maybe I dont understand the intent. When you have telemetry attributes it makes sense to do `AttrsToZapFields` so you log the telemetry. But why would we do it reverse. Do the telemetry `attributes` from `zapcore.Field`.  I would rather extend `log` to include `WithFields` constructor to prevent casting it twice. `fields` -> `attributes` -> `fields`.